tmt/steps/prepare/install.py

```diff
@@ -134,6 +153,52 @@ def enable_copr(self, command, plugin, guest):
                 self.info('copr', copr, 'green')
                 guest.execute(f'{command} copr enable -y {copr}')
 
+    def enable_custom_repo(self, command, guest):
+        """ Enable requested repositories """
+        repos = self.get('repo'):
+        if not repos:
+            return
+
+        # Create repo file content one bye one
+        repo_list = list()
+        for repo_id, repo in enumerate(repos):
+            self.debug(f'Create repo file content for {repo}.')
+            gpgcheck = 1 if self.get('gpgcheck', '0') == '1' else 0
+            skip_if_unavl = 1 if self.get('skip-if-unavailable', '0') == '1'
+                            else 0
+            file_content = dict()
+            file_content['name'] = f'custom-{repo_id}'
+            file_content['baseurl'] = repo
+            file_content['enable'] = 1
+            file_content['gpgcheck'] = gpgcheck
+            file_content['skip_if_unavailable'] = skip_if_unavl
+            repo_list.append(file_content)
```

Comment:
My 2 cents: `configparser.ConfigParser` would make things easier for you, reducing the boilerplate you employ. For our purposes, a yum repo file is an INI file after all, with a single specially named section. Something like this could be perfectly good enough:

```python
import configparser

...

for repo_id, repo in enumerate(repos):
  repo_name = f'custom-{repo_id}'

  repo_config = configparser.ConfigParser()
  repo_config.add_section(repo_name)
  repo_config.set(repo_name, 'name', repo_name)
  repo_config.set(repo_name, 'baseurl', repo)
  repo_config.set(repo_name, 'gpgcheck', '1' if self.get('gpgcheck') else '0')
  ...

  with open(repo_file_local, 'w') as repo_configfile:
    repo_config.write(repo_configfile)
```

Would you mind giving this a try? There should be no need for multiple loops, accumulating `dict`s for each repo, I believe we can easily use a `ConfigParser` object instead and set it up and save in one go.

Reply:
`ConfigParser` works well, which is much better than my  awkward  implementation via `list()`. Thanks very much for your suggestion! @happz